reader-osm/src/test/java/com/graphhopper/routing/RoutingAlgorithmWithOSMTest.java

```diff
@@ -556,201 +511,247 @@ public void testCampoGrande() {
+                                "', got: '" + res.getBest().getDebugInfo() + "'");
             }
         }
+    }
 
-        assertEquals(MAX * algosLength * instances.size(), integ.get());
-        assertEquals(testCollector.toString(), 0, testCollector.errors.size());
-        hopper.close();
+    private void checkResponse(GHResponse res, Query query) {
+        assertFalse(res.hasErrors(), res.getErrors().toString());
+        ResponsePath responsePath = res.getBest();
+        assertFalse(responsePath.hasErrors(), responsePath.getErrors().toString());
+        assertEquals(distCalc.calcDistance(responsePath.getPoints()), responsePath.getDistance(), 2,
+                "responsePath.getDistance does not equal point list distance");
+        assertEquals(query.getPoints().stream().mapToDouble(a -> a.expectedDistance).sum(), responsePath.getDistance(), 2, "unexpected distance");
+        // We check the number of points to make sure we found the expected route.
+        // There are real world instances where A-B-C is identical to A-C (in meter precision).
+        assertEquals(query.getPoints().stream().mapToInt(a -> a.expectedPoints).sum(), responsePath.getPoints().getSize(), 1, "unexpected point list size");
+    }
+
+    private List<Function<Query, GHRequest>> createRequestFactories() {
+        // here we setup different kinds of requests to calculate routes with different algorithms
+        return Arrays.asList(
+                // flex
+                q -> createRequest(q).putHint("expected_algo", "dijkstra-routing")
+                        .putHint("ch.disable", true).putHint("lm.disable", true).setAlgorithm(DIJKSTRA),
+                q -> createRequest(q).putHint("expected_algo", "astar|beeline-routing")
+                        .putHint("ch.disable", true).putHint("lm.disable", true).setAlgorithm(ASTAR),
+                q -> createRequest(q).putHint("expected_algo", "dijkstrabi-routing")
+                        .putHint("ch.disable", true).putHint("lm.disable", true).setAlgorithm(DIJKSTRA_BI),
+                q -> createRequest(q).putHint("expected_algo", "astarbi|beeline-routing")
+                        .putHint("ch.disable", true).putHint("lm.disable", true).setAlgorithm(ASTAR_BI)
+                        .putHint(ASTAR_BI + ".approximation", "BeelineSimplification"),
+                // LM
+                q -> createRequest(q).putHint("expected_algo", "astarbi|landmarks-routing")
+                        .putHint("ch.disable", true)
+                        .setAlgorithm(ASTAR_BI).putHint(ASTAR_BI + ".approximation", "BeelineSimplification"),
+                // CH
+                q -> createRequest(q).putHint("expected_algo", "dijkstrabi|ch-routing")
+                        .setAlgorithm(DIJKSTRA_BI),
+                q -> createRequest(q).putHint("expected_algo", "astarbi|ch-routing")
+                        .setAlgorithm(ASTAR_BI)
+        );
+    }
```

Comment:
Testing the different algorithms is now simply done by using different parameters for `GHRequest`